src/main/java/byteceps/processing/WorkoutLogsManager.java

```diff
@@ -37,6 +37,10 @@ public void addExerciseLog(String workoutLogDate, String exerciseName,
             int setsInt = Integer.parseInt(sets);
             int repsInt = Integer.parseInt(repetitions);
 
+            if (weightInt < 0 || setsInt < 0 || repsInt < 0) {
+                throw new NumberFormatException();
+            }
+
```

Comment:
Good fix for [PE-D][Tester B] Able to insert negative weights when adding a logging a new exercise #115